app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt

```diff
@@ -346,9 +350,31 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    private val REPLAY_SESSION_PATH by lazy {
+        "${application.getExternalFilesDir(null)!!.absolutePath}/Replays/"
+    }
+    private val RECORD_SESSION_PATH by lazy {
+        "${application.getExternalFilesDir(null)!!.absolutePath}/Records/"
+    }
+
     private fun tryToInitVisionManager() {
         if (isPermissionsGranted && !visionManagerWasInit) {
-            VisionManager.create()
+
+            val replay = VisionManager.VisionMode.SessionReplay(
+                path = REPLAY_SESSION_PATH
+            )
+            replay.setSessionProgress(60000)
+
+            val mode = VisionManager.VisionMode.Default(
+                application = application,
+                videoSource = Camera2VideoSourceImpl(application)
+            )
+
+            val recording = VisionManager.VisionMode.SessionRecord(
+                 path = RECORD_SESSION_PATH
+            )
```

Comment:
@deniskoronchik @chezzdev @dersim-davaod this is my initial proposal for API of session recording/replay.

Basically developer can create VisionManager with three possible operating modes, `Default`, `SessionRecord` and `SessionReplay`. 
- `Default` is current mode of operating, with telemetry and videos recorded by sessions.
- `SessionRecord` records complete telemetry session and .mp4 video to `path` specified in constructor.
- `SessionReplay` replays session from session, recorded by previous mode. 

We can extend each mode with different service methods, eg. I've added `setSessionProgress` method to `SessionReplay` mode, which allows to set its progress.

I've added new method for specifying modes, `createWithMode`, and left `create` without changes to be backward compatible. 
Also, we can go with different approach for mode, like different methods for each mode, eg.
`create`, `createRecord`, `createReplay`.